File: htttp/class/AnimalClass.php
<?php

class AnimalClass {
    //-- Propriedades privadas
    private $codAnimal;
    private $Cliente_codCliente;
    private $nome;
    private $anoNascimento;
    private $peso;
    private $grupo;
    private $raca;
    private $genero;
    private $vacinacao;
    private $comportamento;

    //-- Encapsulamento das propriedades (Método GET e SET)
    function getCodAnimal() {
        return $this->codAnimal;
    }

    function getCliente_CodCliente() {
        return $this->Cliente_codCliente;
    }

    function getNome() {
        return $this->nome;
    }

    function getAnoNascimento() {
        return $this->anoNascimento;
    }

    function getPeso() {
        return $this->peso;
    }

    function getGrupo() {
        return $this->grupo;
    }

    function getRaca() {
        return $this->raca;
    }

    function getGenero() {
        return $this->genero;
    }

    function getVacinacao() {
        return $this->vacinacao;
    }

    function getComportamento() {
        return $this->comportamento;
    }

    function setCodAnimal($codAnimal) {
        $this->codAnimal = $codAnimal;
    }

    function setCliente_CodCliente($Cliente_codCliente) {
        $this->Cliente_codCliente = $Cliente_codCliente;
    }

    function setNome($nome) {
        $this->nome = $nome;
    }

    function setAnoNascimento($anoNascimento) {
        $this->anoNascimento = $anoNascimento;
    }

    function setPeso($peso) {
        $this->peso = $peso;
    }


    function setGrupo($grupo) {
        $this->grupo = $grupo;
    }

    function setRaca($raca) {
        $this->raca = $raca;
    }

    function setEndereco($endereco) {
        $this->endereco = $endereco;
    }

    function setGenero($genero) {
        $this->genero = $genero;
    }

    function setVacinacao($vacinacao) {
        $this->vacinacao = $vacinacao;
    }

    function setComportamento($comportamento) {
        $this->comportamento = $comportamento;
    }

    //Métodos CRUD
    public function retAnimais() {
        require_once('class/ConexaoClass.php');
        $objConexao = new ConexaoClass("localhost", "root", "", "dbPrimoPet");
        $tableAnimal = $objConexao->selecionarDados("SELECT * FROM Animal");

        return $tableAnimal;
    }

    public function inserirAnimal($objAnimal) {
        require_once('class/ConexaoClass.php');
        $objConexao = new ConexaoClass("localhost", "root", "", "dbPrimoPet");   
        
        $codAnimal = $objAnimal->getCodAnimal();
        $Cliente_codCliente = $objAnimal->getCliente_CodCliente();
        $nome = $objAnimal->getNome();
        $anoNascimento = $objAnimal->getAnoNascimento();
        $peso = $objAnimal->getPeso();
        $grupo = $objAnimal->getGrupo();
        $raca = $objAnimal->getRaca();
        $genero = $objAnimal->getGenero();
        $vacinacao = $objAnimal->getVacinacao();
        $comportamento = $objAnimal->getComportamento();
        
        $objConexao->exercutarComandoSQL("INSERT INTO Animal (codAnimal, Cliente_codCliente, nome, anoNascimento, peso, grupo, raca, genero, vacinacao, comportamento) VALUES ('$codAnimal', '$Cliente_codCliente', '$nome', '$anoNascimento', '$peso', '$grupo', '$raca', '$genero', '$vacinacao', '$comportamento')");               

        return true;
    }

    public function editarAnimal($objAnimal) {
        require_once('class/ConexaoClass.php');
        $objConexao = new ConexaoClass("localhost", "root", "", "dbPrimoPet");   
        
        $codAnimal = $objAnimal->getCodAnimal();
        $Cliente_codCliente = $objAnimal->getCliente_CodCliente();
        $nome = $objAnimal->getNome();
        $anoNascimento = $objAnimal->getAnoNascimento();
        $peso = $objAnimal->getPeso();
        $grupo = $objAnimal->getGrupo();
        $raca = $objAnimal->getRaca();
        $genero = $objAnimal->getGenero();
        $vacinacao = $objAnimal->getVacinacao();
        $comportamento = $objAnimal->getComportamento();
       
        $objConexao->exercutarComandoSQL("UPDATE Animal SET codAnimal='$codAnimal', Cliente_codCliente='$Cliente_codCliente', nome='$nome',anoNascimento='$anoNascimento', peso='$peso', grupo='$grupo', raca='$raca', genero='$genero', vacinacao='$vacinacao', comportamento='$comportamento' WHERE codAnimal='$codAnimal'");               

        return true;
    }

    public function excluirAnimal($codAnimal) {
        require_once('class/ConexaoClass.php');
        $objConexao = new ConexaoClass("localhost", "root", "", "dbPrimoPet");   
        
        $objConexao->exercutarComandoSQL("DELETE FROM Animal WHERE codAnimal='$codAnimal'");               

        return true;
    }

    public function retProxCodAnimal(){
        $tableclientes = $this->retAnimais();
        $max = sizeof($tableclientes);
        $codAntigo  = $tableclientes[$max-1]['codAnimal'];
        $codAntigo = substr($codAntigo,4,3);
        $numNovo =  intval($codAntigo)+1;
        $numString = (string)$numNovo;
        while (strlen($numString) < 3){
            $numString = '0'.$numString;
        }
        $codNovo = 'ANI'.$numString;
        return $codNovo;
    }
}

File: ProjetoDeAplicativos/ProjetoDeAplicativos/AnimalClass.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data;

namespace ProjetoDeAplicativos
{
    class AnimalClass
    {

        public string codAnimal { get; set; }
        public string Cliente_codCliente { get; set; }
        public string nome { get; set; }
        public string anoNascimento { get; set; }
        public string peso { get; set; }
        public string grupo { get; set; }
        public string raca { get; set; }
        public string genero { get; set; }
        public string vacinacao { get; set; }
        public string comportamento { get; set; }

        public DataTable retAnimais()
        {
            ClasseDB db = new ClasseDB();
            return db.RetornarTabela("SELECT * FROM Animal");
        }

        public void inserirAnimal()
        {
            ClasseDB db = new ClasseDB();
            db.ExecutarComando($"INSERT INTO Animal (codAnimal, Cliente_codCliente, nome, anoNascimento, peso, grupo, raca, genero, vacinacao, comportameto) VALUES ('{codAnimal}', '{Cliente_codCliente}', '{nome}', '{anoNascimento}', '{peso}', '{grupo}', '{raca}', '{genero}', '{vacinacao}', '{comportamento}')");
        }

        public void editarAnimal()
        {
            ClasseDB db = new ClasseDB();
            db.ExecutarComando($"UPDATE Animal SET codAnimal='{codAnimal}', Cliente_codCliente='{Cliente_codCliente}', nome='{nome}',anoNascimento='{anoNascimento}', peso='{peso}', grupo='{grupo}', raca='{raca}', genero='{genero}', vacinacao='{vacinacao}', comportameto='{comportamento}' WHERE codAnimal='{codAnimal}'");
        }

        public void excluirAnimal(string codAnimal)
        {
            ClasseDB db = new ClasseDB();
            db.ExecutarComando($"DELETE FROM Animal WHERE codAnimal='{codAnimal}'");
        }

        public void excluirAnimal()
        {
            ClasseDB db = new ClasseDB();
            db.ExecutarComando($"DELETE FROM Animal WHERE codAnimal='{codAnimal}'");
        }

        public string retProxCodAnimal()
        {
            DataTable tabelaanimais = retAnimais();
            int max = tabelaanimais.Rows.Count;
            string codAntigo = tabelaanimais.Rows[max - 1]["'codAnimal'"].ToString();
            codAntigo = codAntigo.Substring(4, 3);
            int numNovo = int.Parse(codAntigo) + 1;
            string numString = numNovo.ToString();
            while (numString.Length < 3)
            {
                numString = "0" + numString;
            }
            string codNovo = "ANI" + numString;
            return codNovo;
        }
    }
}
